drivers/frequency/ad9528/ad9528.c

```diff
@@ -344,6 +355,417 @@ int32_t ad9528_init(struct ad9528_init_param *init_param)
 	return(0);
 }
 
+/**
+ * @brief Initialize the CLK structure.
+ *
+ * @param desc - The CLK descriptor.
+ * @param init_param - The structure holding the device initial parameters.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int ad9528_clk_init(struct no_os_clk_desc **desc,
+			   const struct no_os_clk_init_param *init_param)
```

Comment:
no_os_clk_init() takes care of the memory allocation, so, this function can be removed.

Reply:
Fixed.